src/metaswitch/clearwater/plugin_tests/test_cassandra_failed_plugin.py

```diff
@@ -0,0 +1,86 @@
+# @file test_failed_cassandra_plugin.py
+#
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2016 Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import unittest
+import mock
+import logging
+
+_log = logging.getLogger()
+
+from clearwater_etcd_plugins.clearwater_cassandra.cassandra_failed_plugin import CassandraFailedPlugin
+from metaswitch.clearwater.cluster_manager.plugin_base import PluginParams
+from metaswitch.clearwater.cluster_manager import alarm_constants
+from metaswitch.clearwater.cluster_manager.plugin_utils import WARNING_HEADER
+
+class TestCassandraFailedPlugin(unittest.TestCase):
+    # run_command returns 0 if a command completes successfully, but python mocks
+    # return 'True', i.e. 1. Force return value of 0 to simulate successes.
+    @mock.patch('clearwater_etcd_plugins.clearwater_cassandra.cassandra_failed_plugin.run_command',\
+                return_value=0)
+    @mock.patch('metaswitch.common.alarms.alarm_manager.get_alarm')
+    # The plugin uses check_output to get the latency value, so we return 100000.
```

Comment:
The failed plugin doesn't do check_output on the latency, so I don't think we need it.